src/main/java/htsjdk/samtools/cram/index/CRAIIndex.java

```diff
@@ -96,59 +87,49 @@ public static SeekableStream openCraiFileAsBaiStream(final InputStream indexStre
             slice.alignmentStart = entry.getAlignmentStart();
             slice.alignmentSpan = entry.getAlignmentSpan();
             slice.sequenceId = entry.getSequenceId();
-            // NOTE: the recordCount and sliceIndex fields can't be derived from the CRAM index
+            slice.offset = entry.getSliceByteOffset();
+
+            // NOTE: the sliceIndex and read count fields can't be derived from the CRAM index
             // so we can only set them to zero
             // see https://github.com/samtools/htsjdk/issues/531
-            slice.nofRecords = 0;
+
             slice.index = 0;
-            slice.offset = entry.getSliceByteOffset();
+            slice.mappedReadsCount = 0;
+            slice.unmappedReadsCount = 0;
+            slice.unplacedReadsCount = 0;
 
-            indexer.processSingleReferenceSlice(slice);
+            indexer.processAsSingleReferenceSlice(slice);
         }
         indexer.finish();
 
         return new SeekableMemoryStream(baos.toByteArray(), "CRAI to BAI converter");
     }
 
     public static List<CRAIEntry> find(final List<CRAIEntry> list, final int seqId, final int start, final int span) {
-        final boolean whole = start < 1 || span < 1;
-        final CRAIEntry query = new CRAIEntry(seqId,
-                start < 1 ? 1 : start,
```

Comment:
these ternary operators are useless, since `whole` (now `matchEntireSequence`) handles this case